synapse/tests/test_lib_kv.py

```diff
@@ -0,0 +1,240 @@
+
+                # We can overwrite a key
+                self.none(stor.setKvProp(key, valu + valu))
+                self.eq(stor.getKvProp(key), valu + valu)
+
+                # Missing keys are None
+                self.none(stor.getKvProp(valu))
+
+                # We can delete a key
+                self.true(stor.delKvProp(key))
+                self.false(stor.delKvProp(key))
+
+                # We can set a bunch of values in a single xact
+                self.none(stor.setKvProps(propd))
+                self.eq(stor.getKvProp(b'prefix:1'), b'woah')
+                self.eq(stor.getKvProp(b'prefix:2'), b'dude')
+                self.eq(stor.getKvProp(b'pennywise:derry'), b'georgie')
+
+                # We can iterate over a given prefix of data
+                vals = [tup for tup in stor.iterKvProps(b'prefix:')]
+                self.len(2, vals)
+                ed = propd.copy()
+                ed.pop(b'pennywise:derry')
+                self.eq(ed, dict(vals))
+
+                # We may get no items during iteration too
+                vals = [tup for tup in stor.iterKvProps(b'notaprefix:')]
+                self.eq(vals, [])
+
+                # Some keys may have multiple values in them via dup API
+                self.false(stor.hasKvDups(dupkey))
+                self.none(stor.addKvDup(dupkey, valu))
+                self.none(stor.addKvDup(dupkey, valu + valu))
+                self.none(stor.addKvDup(dupkey, valu + valu + valu))
+                self.true(stor.hasKvDups(dupkey))
+                vals = [tup for tup in stor.iterKvDups(dupkey)]
+                self.eq(vals, [valu, valu + valu, valu + valu + valu])
+
+                # We can delete a single value from a dup list
+                self.true(stor.delKvDup(dupkey, valu + valu))
+                self.false(stor.delKvDup(dupkey, valu + valu))
+                vals = [tup for tup in stor.iterKvDups(dupkey)]
+                self.eq(vals, [valu, valu + valu + valu])
+
+                # We can set a bunch of dup props at once
+                self.none(stor.addKvDups(duplist))
+                # duplist contains one duplicate key/valu tuple,
+                # which is discarded by lmdb
+                self.len(6, duplist)
+                self.len(3, [val for val in stor.iterKvDups(dlk1)])
+                self.len(2, [val for val in stor.iterKvDups(dlk2)])
+
+                # We can generate a named iden to use as an alias
+                iden = stor.genKvAlias('vertex')
+                self.len(16, iden)
+                self.isinstance(iden, bytes)
+                # the aliases are still stable
+                niden = stor.genKvAlias('vertex')
+                self.eq(iden, niden)
+                another_iden = stor.genKvAlias('pennywise')
+                self.len(16, another_iden)
+
+                # We cannot have keys which are empty bytes
+                self.raises(lmdb.BadValsizeError, stor.setKvProp, b'', b'newp')
```

Comment:
Should we raise our own exception here instead of letting lmdb errors bubble up?